usr/lib/icsf_stdll/icsf.c

```diff
@@ -2976,6 +2976,9 @@ int icsf_hash_signverify(LDAP * ld, int *reason, struct icsf_object_record *key,
         return -1;
     }
 
+    if (sig == NULL && sig_len != NULL)
+        *sig_len = 0;
```

Comment:
shouldnt we return here directly in case of size query ?

Reply:
No, this would not be a real size query then, as it would always return zero as size, but that's not what is desired. 

The code further down (i.e. icsf_call) will do the real size query, if the (ber encoded) signature length is zero (or smaller than required for the signature) in the request. The result returned from icsf_call will then contain the required size, and *sig_len is set from that. 

Setting *sig_len to zero is required, because *sig_len could contain any value on entry. So by setting *sig_len to zero first (if sig is NULL) we ensure that a size query is done, regardless what *sig_len was on entry.